client/src/services/data-interface/data-interface.ts

```diff
@@ -190,26 +204,40 @@ export default function getNumOfLicenses(
   }
 
   if (options?.filterByAlcoholType && options?.filterByZipcode) {
-    const licenseByZipAndType = data.filter(
+    const licensesByZipAndType = data.filter(
       (license) =>
         license.zipcode === options.filterByZipcode &&
         license.alcohol_type === options.filterByAlcoholType
     );
 
-    return licenseByZipAndType.length;
+    if (options.filterByAlcoholType === "All Alcoholic Beverages") {
+      return MAX_ALL_ALC_PER_ZIP - licensesByZipAndType.length;
+    } else if (options.filterByAlcoholType === "Wines and Malt Beverages") {
+      return MAX_BEER_WINE_PER_ZIP - licensesByZipAndType.length;
+    } else {
+      console.error("improper alcohol license type used");
+      return -1;
+    }
```

Comment:
🌴 🌴 🌴 Super-non-blocking comment: Control flow is starting to get a bit gnarly here. I wouldn't necessarily refactor it now though - probably not worth the effort.

If you come back here later, consider how you might refactor to reduce complex nested if statements. Especially do this if you find yourself introducing more deeply nested if statements.